sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -1097,4 +1097,9 @@ private[spark] object QueryCompilationErrors {
       s"and column $colName cannot be resolved. Expected $expectedNum columns named $colName but " +
       s"got ${actualCols.map(_.name).mkString("[", ",", "]")}")
   }
+
+  def numberOfPartitionsNotAllowedWithUnspecifiedDistributionError(): Throwable = {
+    throw new AnalysisException("The number of partitions can't be specified with unspecified" +
+      " distribution. Invalid Sink requirements detected.")
```

Comment:
nit: why "Invalid Sink requirements detected"? is this error specific to streaming?

Reply:
Nope. I thought the word "Sink" is not specific to streaming, but looks like it brings confusion. Just changed to "writer". Thanks!